app/src/main/java/com/huji/foodtricks/buddies/ViewSingleEventActivity.java

```diff
@@ -142,20 +141,38 @@ public void onUpdateSuccess() {
             }
 
         });
+        modifyRSVPButtons();
+    }
+
+    private void modifyRSVPButtons() {
+        Set<Integer> allRSVPButtons = new HashSet<>(ALL_RSVP_BUTTONS);
+        for (int buttonId : allRSVPButtons) {
+            Button currButton = findViewById(buttonId);
+            changeButtonToDisabled(currButton);
+        }
         TextView rsvp_tv = findViewById(R.id.RSVPText);
         modifyAttendersTextView(curr_event.getAttendanceProvider(), rsvp_tv);
-        //TODO: refactor the following code
-        Set<Integer> allRSVPButtons = new HashSet<>(Arrays.asList(R.id.approve_btn, R.id.tentative_btn, R.id.decline_btn));
-        allRSVPButtons.remove(view.getId()); // remove the selected button from the list of buttons to disable
-        Button selectedButtonView = (Button) view; // just for readability
+        EventAttendanceProvider attendanceProvider = curr_event.getAttendanceProvider();
+        EventAttendanceProvider.RSVP status = attendanceProvider.getUserRSVP(currentUserID);
+        if (status == EventAttendanceProvider.RSVP.ATTENDING) {
+            changeButtonToEnabled(findViewById(R.id.approve_btn));
+        }
+        else if (status == EventAttendanceProvider.RSVP.NOT_ATTENDING)
+            changeButtonToEnabled(findViewById(R.id.decline_btn));
+        else if (status == EventAttendanceProvider.RSVP.TENTATIVE)
+            changeButtonToEnabled(findViewById(R.id.tentative_btn));
+
+
+
+    }
+
+    private void changeButtonToEnabled(Button selectedButtonView) {
         selectedButtonView.setBackgroundColor(getResources().getColor(R.color.selectedRSVPButton));
         selectedButtonView.setTextColor(Color.WHITE);
-//        setContentView(R.layout.activity_view_single_event);
+    }
 
-        for (int buttonId : allRSVPButtons) {
-            Button currButton = findViewById(buttonId);
-            currButton.setBackgroundColor(Color.WHITE);
-            currButton.setTextColor(Color.BLACK);
-        }
+    private void changeButtonToDisabled(Button currButton) {
+        currButton.setBackgroundColor(Color.WHITE);
+        currButton.setTextColor(Color.BLACK);
```

Comment:
you should also actually disable the button (and enable it in "changeToEnabled") so the user can't click it again..